NYT.py

```diff
@@ -0,0 +1,213 @@
+    StructField('snippet', StringType(), True), 
+    StructField('lead_paragraph', StringType(), True), 
+    StructField('print_section', StringType(), True), 
+    StructField('print_page', FloatType(), True), 
+    StructField('source', StringType(), False), 
+    StructField('multimedia', StringType(), True), 
+    StructField('headline', StringType()), 
+    StructField('keywords', StringType(), True),
+    StructField('pub_date', TimestampType(), False), 
+    StructField('document_type', StringType(), True), 
+    StructField('news_desk', StringType(), True), 
+    StructField('section_name', StringType(), True), 
+    StructField('byline', StringType(), True),
+    StructField('type_of_material', StringType(), True), 
+    StructField('_id', StringType(), False), 
+    StructField('word_count', FloatType(), True), 
+    StructField('uri', StringType(), False), 
+    StructField('subsection_name', StringType(), True),
+    StructField('corrupted', StringType(), True)
+])
+
+nyt_articles = (spark
+    .read
+    .option('header', 'true')
+    .option("quote", "\"")
+    .option("escape", "\"")
+    .option("timestampFormat", "yyyy-MM-dd HH:mm:ss") \
+    .schema(nyt_schema)
+    .csv('/FileStore/nyt-metadata.csv', multiLine=True))
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ###Showing the DataFrame
+
+# COMMAND ----------
+
+from pyspark.sql.functions import rand 
+display(nyt_articles.orderBy(rand()).limit(10))
+nyt_articles.printSchema()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Repairing data
+
+# COMMAND ----------
+
+from pyspark.sql.functions import col, to_timestamp, levenshtein
+
+#there is a row that is split into two with null values 
+row1 = nyt_articles.filter(col("web_url") == "https://www.nytimes.com/2022/04/28/sports/football/nfl-draft-hats-2022.html")
+row2 = nyt_articles.filter(col("section_name") == "nyt://article/2aec1dad-e9d0-5a3c-931b-cc93890bb60e")
+
+row1 = row1.select(nyt_articles.columns[:3])
+row2 = row2.select(nyt_articles.columns[:18])
+combined_row = row1.crossJoin(row2)
+combined_row = combined_row.toDF(*nyt_articles.columns)
+display(combined_row)
+
+# COMMAND ----------
+
+#replace 2 broken rows with repaired one
+nyt_articles = nyt_articles.filter(col("web_url") != "https://www.nytimes.com/2022/04/28/sports/football/nfl-draft-hats-2022.html")
+nyt_articles = nyt_articles.filter(col("section_name") != "nyt://article/2aec1dad-e9d0-5a3c-931b-cc93890bb60e")
+nyt_articles = nyt_articles.union(combined_row)
+
+# COMMAND ----------
+
+#after reading digital values as floats, we should change them to integer because every single one is .0
+nyt_articles = nyt_articles.withColumn("print_page", col("print_page").cast(IntegerType()))
+nyt_articles = nyt_articles.withColumn("word_count", col("word_count").cast(IntegerType()))
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ###Checking for errors and cleaning data
+
+# COMMAND ----------
+
+def error_checker(df, filtered_df):
```

Comment:
I would move all functions to a separate file and import from there. This way you would not distract from code